### ISSUE TITLE
[require-array-sort-compare] add option to ignore string arrays

### ISSUE THREAD

**Issue by finnigantime** [typescript-eslint/typescript-eslint#1963]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/require-array-sort-compare": []
  }
}
```

```TS
['c', 'b', 'a'].sort();
```

**Expected Result**
no error

**Actual Result**
`error: Require 'compare' argument. [@typescript-eslint/require-array-sort-compare]`

**Additional Info**
In the original rule implementation PR, there's a comment that an `ignoreStringArrays` option would be useful but can be punted on until someone asks for it: https://github.com/typescript-eslint/typescript-eslint/pull/261#issuecomment-463439800

I'm asking for it :)

The spirit of the rule makes sense but it's cumbersome to sprinkle comparator functions across lots of test files that are just sorting string arrays for the purposes of stable deep comparison. Why should we have to provide a comparator function for sorting a string array? The language handles this natively for most cases without the need to supply a custom function.

I would like to have the `ignoreStringArrays` option. And IMO it would be nice to set that to true by default, but I understand if for back-compat or for purposes of having devs think about edge cases like cross-locale comparison it makes sense to keep this option off by default.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.27.0` |
| `@typescript-eslint/parser`        | `2.27.0` |
| `TypeScript`                       | `3.6.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.12.0` |
